nemo/collections/tts/models/squeezewave.py

```diff
@@ -0,0 +1,212 @@
+            'loss': loss,
+            'progress_bar': {'training_loss': loss},
+            'log': {'loss': loss},
+        }
+
+    def validation_step(self, batch, batch_idx):
+        self.mode = OperationMode.validation
+        self.squeezewave.mode = OperationMode.validation
+        audio, audio_len = batch
+        z, log_s_list, log_det_W_list, audio_pred, spect, spect_len = self.forward(
+            audio=audio, audio_len=audio_len, run_inverse=(batch_idx == 0)
+        )
+        loss = self.loss(z=z, log_s_list=log_s_list, log_det_W_list=log_det_W_list, sigma=self.sigma)
+        return {
+            "val_loss": loss,
+            "audio_pred": audio_pred,
+            "mel_target": spect,
+            "mel_len": spect_len,
+        }
+
+    def validation_epoch_end(self, outputs):
+        if self.logger is not None and self.logger.experiment is not None:
+            waveglow_log_to_tb_func(
+                self.logger.experiment,
+                outputs[0].values(),
+                self.global_step,
+                tag="eval",
+                mel_fb=self.audio_to_melspec_precessor.fb,
+            )
+        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
+        tensorboard_logs = {'val_loss': avg_loss}
+        return {'val_loss': avg_loss, 'log': tensorboard_logs}
+
+    def __setup_dataloader_from_config(self, cfg, shuffle_should_be: bool = True, name: str = "train"):
+        if "dataset" not in cfg or not isinstance(cfg.dataset, DictConfig):
+            raise ValueError(f"No dataset for {name}")  # TODO
+        if "dataloader_params" not in cfg or not isinstance(cfg.dataloader_params, DictConfig):
+            raise ValueError(f"No dataloder_params for {name}")  # TODO
+        if shuffle_should_be:
+            if 'shuffle' not in cfg.dataloader_params:
+                logging.warning(
+                    f"Shuffle should be set to True for {self}'s {name} dataloader but was not found in its "
+                    "config. Manually setting to True"
+                )
+                with open_dict(cfg["dataloader_params"]):
+                    cfg.dataloader_params.shuffle = True
+            elif not cfg.dataloader_params.shuffle:
+                logging.error(f"The {name} dataloader for {self} has shuffle set to False!!!")
+        elif not shuffle_should_be and cfg.dataloader_params.shuffle:
+            logging.error(f"The {name} dataloader for {self} has shuffle set to True!!!")
+
+        dataset = instantiate(cfg.dataset)
+        return torch.utils.data.DataLoader(dataset, collate_fn=dataset.collate_fn, **cfg.dataloader_params)
+
+    def setup_training_data(self, cfg):
+        self._train_dl = self.__setup_dataloader_from_config(cfg)
+
+    def setup_validation_data(self, cfg):
+        self._validation_dl = self.__setup_dataloader_from_config(cfg, shuffle_should_be=False, name="validation")
+
+    @classmethod
+    def list_available_models(cls) -> 'Optional[Dict[str, str]]':
```

Comment:
Please give a checkpoint to @okuchaiev, so we can populate this method

Reply:
yes, please. You can produce ".nemo" file by calling .save_to(...) on a trained model